src/tests/tests/background.js

```diff
@@ -119,4 +124,22 @@
     }
   });
 
+  QUnit.test("DNT checking obeys user setting", (assert) => {
+    const NUM_TESTS = 5;
+
+    let done = assert.async(NUM_TESTS);
+
+    assert.expect(NUM_TESTS);
+
+    badger.isCheckingDNTPolicyEnabled = () => false;
```

Comment:
Agree that the fact that unit tests can modify local settings when run against your local Badger is a bug. It would be good to get it fixed, but it's not a blocker for this pull request.